subprojects/internal-testing/src/main/groovy/org/gradle/util/SetSystemProperties.java

```diff
@@ -53,41 +47,12 @@ public Statement apply(final Statement base, Description description) {
             @Override
             public void evaluate() throws Throwable {
                 System.getProperties().putAll(customProperties);
```

Comment:
Should we check that we're not trying to override the tmp dir here so more tests aren't introduced that rely on it?

Looking at https://github.com/gradle/gradle/blob/b3f9768cce6dd5d3a6a91daeec23998130c43898/subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java#L989-L997 - it seems like we always use a non-standard tmpdir.  This works for the forking tests, but I wonder if we should just pass java.io.tmpdir as a system property to the test worker?

Reply:
Per our chat, I've added some logic to throw an exception if you try to set "java.io.tmpdir" but we'll leave any effort to change the tmpdir of the test worker to later if we run into issues around it (i.e. setting it creates other issues and it's probably not worth the effort right now to resolve those since we aren't feeling any pain by it not being set).